mapviz_plugins/src/point_drawing_plugin.cpp

```diff
@@ -0,0 +1,437 @@
+// *****************************************************************************
+//
+// Copyright (c) 2014, Southwest Research Institute® (SwRI®)
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Southwest Research Institute® (SwRI®) nor the
+//       names of its contributors may be used to endorse or promote products
+//       derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL <COPYRIGHT HOLDER> BE LIABLE FOR ANY
+// DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+// (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+// LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+// ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// *****************************************************************************
+
+#include <mapviz_plugins/point_drawing_plugin.h>
+
+#include <vector>
+#include <list>
+
+#include <QDialog>
+#include <QGLWidget>
+#include <QPalette>
+#include <QPainter>
+
+#include <opencv2/core/core.hpp>
+
+#include <swri_image_util/geometry_util.h>
+#include <swri_transform_util/transform_util.h>
+
+namespace mapviz_plugins
+{
+  PointDrawingPlugin::PointDrawingPlugin()
+      : new_lap_(true),
+        got_begin_(false),
+        draw_style_(LINES),
+        position_tolerance_(0.0),
+        lap_checked_(false)
```

Comment:
This class has a few primitive types that you are not initializing.  It would probably be a good idea to also set:
```
int buffer_size_;
bool covariance_checked_;
int buffer_holder_
```

Also make sure they're initialized here in the same order that they're declared in the .h file.